explainability-service/src/main/java/org/kie/trustyai/service/prometheus/PrometheusScheduler.java

```diff
@@ -15,43 +15,51 @@
 import org.kie.trustyai.service.config.ServiceConfig;
 import org.kie.trustyai.service.data.DataSource;
 import org.kie.trustyai.service.data.exceptions.DataframeCreateException;
-import org.kie.trustyai.service.endpoints.metrics.MetricsCalculator;
-import org.kie.trustyai.service.payloads.ReconciledMetricRequest;
+import org.kie.trustyai.service.endpoints.metrics.MetricsDirectory;
+import org.kie.trustyai.service.payloads.metrics.BaseMetricRequest;
+import org.kie.trustyai.service.payloads.metrics.RequestReconciler;
 
 import io.quarkus.scheduler.Scheduled;
 
 @Singleton
 public class PrometheusScheduler {
 
     private static final Logger LOG = Logger.getLogger(PrometheusScheduler.class);
-    private final ConcurrentHashMap<UUID, ReconciledMetricRequest> spdRequests = new ConcurrentHashMap<>();
-    private final ConcurrentHashMap<UUID, ReconciledMetricRequest> dirRequests = new ConcurrentHashMap<>();
+    private final ConcurrentHashMap<String, ConcurrentHashMap<UUID, BaseMetricRequest>> requests = new ConcurrentHashMap<>();
     @Inject
     Instance<DataSource> dataSource;
     @Inject
     PrometheusPublisher publisher;
-    @Inject
-    MetricsCalculator calculator;
 
     @Inject
     ServiceConfig serviceConfig;
 
-    public Map<UUID, ReconciledMetricRequest> getDirRequests() {
-        return dirRequests;
+    private final MetricsDirectory metricsDirectory = new MetricsDirectory();
+
+    public MetricsDirectory getMetricsDirectory() {
+        return metricsDirectory;
     }
 
-    public Map<UUID, ReconciledMetricRequest> getSpdRequests() {
-        return spdRequests;
+    public Map<UUID, BaseMetricRequest> getRequests(String metricName) {
+        if (requests.containsKey(metricName)) {
+            return this.requests.get(metricName);
+        } else {
+            return new ConcurrentHashMap<>();
+        }
```

Comment:
Is it equivalent to this?
```suggestion
        return this.requests.getOrDefault(metricName, new ConcurrentHashMap<>());
```